From 201551e60fd61a00d27e5eb3e7c8a9391453bb38 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Thu, 27 Apr 2017 12:17:28 -0700
Subject: [PATCH 0635/1239] BGX: Fix QSGMII on T81 alternative package

No need to do anything special for initializing qlm[] for alternative
package, it will be sorted out later.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/net/cavium/thunder_bgx.c | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/drivers/net/cavium/thunder_bgx.c b/drivers/net/cavium/thunder_bgx.c
index 6bbc2736b5..55bb5f9955 100644
--- a/drivers/net/cavium/thunder_bgx.c
+++ b/drivers/net/cavium/thunder_bgx.c
@@ -1352,17 +1352,12 @@ int thunderx_bgx_probe(struct udevice *dev)
 		/* BGX3 (DLM4), has only 2 lanes */
 		if (CAVIUM_IS_MODEL(CAVIUM_CN83XX) && (bgx_idx == 3) && lmac >= 2)
 			continue;
-		if (is_altpkg && (lmac == 2) && (bgx_idx == 0)) {
-			qlm[lmac - 1] = get_qlm_for_bgx(node, bgx_idx, lmac);
-			debug("qlm[%d] = %d\n", lmac, qlm[lmac-1]);
-		} else {
-			qlm[lmac + 0] = get_qlm_for_bgx(node, bgx_idx, lmac);
-			/* Each DLM has 2 lanes, configure both lanes with
-			   same qlm configuration */
-			if (inc == 2)
-				qlm[lmac + 1] = qlm[lmac];
-			debug("qlm[%d] = %d\n", lmac, qlm[lmac]);
-		}
+		qlm[lmac + 0] = get_qlm_for_bgx(node, bgx_idx, lmac);
+		/* Each DLM has 2 lanes, configure both lanes with
+		   same qlm configuration */
+		if (inc == 2)
+			qlm[lmac + 1] = qlm[lmac];
+		debug("qlm[%d] = %d\n", lmac, qlm[lmac]);
 	}
 
 	/* A BGX can take 1 or 2 DLMs, if both the DLMs are not configured
-- 
2.29.0

